cognite_toolkit/_cdf_tk/storageio/_instances.py

```diff
@@ -54,10 +57,93 @@ class InstanceIO(
         return f"{item.space}:{item.external_id}"
 
+    def _get_restricted_properties_for_view(self, view_id: ViewId) -> set[str]:
+        """Get the set of restricted property names for a given view.
+
+        This checks if the view writes to the CogniteAsset container in the cdf_cdm space,
+        and if so, returns the view properties that should be filtered out during upload.
+
+        The CogniteAsset container has managed properties for asset hierarchy that should not
+        be directly written:
+        - pathLastUpdatedTime (maps to assetHierarchy_path_last_updated_time)
+        - path (maps to assetHierarchy_path)
+        - root (maps to assetHierarchy_root)
+
+        Args:
+            view_id: The ViewId to check
+
+        Returns:
+            Set of property names that should be filtered from upload data
+        """
+        # Return cached result if available
+        if view_id in self._view_restricted_properties_cache:
+            return self._view_restricted_properties_cache[view_id]
+
+        restricted_properties: set[str] = set()
+
+        # Container properties that are read-only should not be uploaded.
+        RESTRICTED_CONTAINER_PROPERTIES = {
+            ContainerId(space="cdf_cdm", external_id="CogniteAsset"): {
+                "assetHierarchy_path_last_updated_time",
+                "assetHierarchy_path",
+                "assetHierarchy_root",
+            },
+            ContainerId(space="cdf_cdm", external_id="CogniteFile"): {"isUploaded"},
+        }
+
+        # Retrieve the view to check its properties
+        view = self.client.data_modeling.views.retrieve(view_id)
+        if view is None:
+            # Cache empty result
+            self._view_restricted_properties_cache[view_id] = restricted_properties
+            return restricted_properties
+        view = view[-1]
+
+        # Check each property in the view
+        for property_identifier, property in view.properties.items():
+            if isinstance(
+                property, MappedProperty
+            ) and property.container_property_identifier in RESTRICTED_CONTAINER_PROPERTIES.get(
+                property.container, set()
+            ):
+                restricted_properties.add(property_identifier)
+
+        # Cache the result
+        self._view_restricted_properties_cache[view_id] = restricted_properties
+        return restricted_properties
+
+    def _filter_restricted_properties(self, item_json: dict[str, JsonVal], selector: InstanceSelector | None) -> None:
+        """
+        Filter out restricted properties from the item JSON based on the selector's view.
+
+        Args:
+            item_json: The item data as a JSON dictionary
+            selector: The selector containing view information
+        """
+        if (
+            selector is None
+            or not isinstance(selector, InstanceViewSelector | InstanceSpaceSelector)
+            or selector.view is None
+        ):
+            return
```

Comment:
This is a bit shaky. If the user uploads from a file without any selector you do not filter out properties. Suggest to check the sources instead.